src/main.cpp

```diff
@@ -92,6 +94,14 @@ static void lPrintVersion() {
 #endif
 }
 
+
+// Command line argument constants
+static constexpr std::string_view ARG_NAME_ONLY_PREPROCESSOR{"--onlycpp"};
```

Comment:
What's the point of having this as a `string_view` vs just using the string literal in the `printf()`?

Reply:
I like this style (explicitly named constants) for a few reasons:
* It is portable (uses `std`),
* It is resolved at compile-time,
* It leverages type info/behavior, for example the calls `strncmp` which are safer than calls to `strcmp`,
* It reduces the line size and clutter of the conditional logic in the body (max number of line characters is determined by naming convention and not the string contents),
* It allows for better IDE interactions (e.g. typing `ARG_` is enough to autocomplete contextually when filling out the conditional logic).

Happy to change this back to inline literals.

Reply:
My main concern is readability of resulting code, i.e. right now it's almost 1:1 what we have in the source file vs the output of the `--help`. With ARG_* thing, it's difficult to see the complete picture and catch obvious things looking at source code.

`strcmp`/`strncmp` argument is fair, though for relatively short things, like `-E`, it's less important.

As description string is used only once in the code and parameter name only twice (in the help message and in the matching logic), adding the wrappers will significantly increase code size (in terms of source lines / number of characters in the source, not executable size) and will compromise readability.

So, I suggest doing it old way. Or consider redoing the whole thing, if/when we get too many switches, so readability of the help message / source code will not matter that much.